src/ORM/Hierarchy/Hierarchy.php

```diff
@@ -31,6 +33,8 @@
  */
 class Hierarchy extends Extension
 {
+    use Configurable;
```

Comment:
Add Configurable here is a little strange in that there are already a bunch of private static's on this class, however it's an Extension class so prior to this class the config was being read elusively from the DataObjects the Extension was applied to

I've added Configurable here because this is the most logical place central place to house the default_children_method config which is used by 3 other classes (2x in this PR, 1x in the silverstripe/cms PR)

Reply:
We need to keep the oddity, there needs to be a default somewhere.  Otherwise the config must be defined on SiteTree and anything else that uses Hierarchy.

> Otherwise we'll need to specify in PHPDocs of each config whether they're for use in the extension or in the class being extended

I don't quite understand what this means.  There's nothing stopping people from using either `Hierarchy::config()->get('default_children_method');` or `SiteTree::config()->get('default_children_method');`

Reply:
We don't need to use the `Configurable` trait. The new configuration is no different to the others already defined in this extension such as `allowed_children` or `can_be_root`.

```suggestion
```

Reply:
Updated